core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/HandleResponseStage.java

```diff
@@ -109,14 +109,16 @@ private OutputT handleSuccessResponse(SdkHttpFullResponse httpResponse, RequestE
      * @throws IOException If any problems are encountering reading the error response.
      */
     private SdkException handleErrorResponse(SdkHttpFullResponse httpResponse,
-                                             RequestExecutionContext context)
-            throws IOException, InterruptedException {
+                                             RequestExecutionContext context) throws IOException, InterruptedException {
         try {
             SdkException exception = errorResponseHandler.handle(httpResponse, context.executionAttributes());
             exception.fillInStackTrace();
             SdkStandardLogger.REQUEST_LOGGER.debug(() -> "Received error response: " + exception);
             return exception;
-        } catch (InterruptedException | IOException e) {
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw e;
+        } catch (IOException e) {
```

Comment:
In what circumstance would the thread not have already been interrupted when interrupted exception is thrown?